ledger-core/insolar/payload/validation.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package payload
+
+import (
+	"github.com/insolar/assured-ledger/ledger-core/vanilla/throw"
+)
+
+type Validate interface {
+	Validate(currPulse PulseNumber) error
+}
+
+var _ Validate = &VStateReport{}
+
+func (m *VStateReport) Validate(currPulse PulseNumber) error {
+	if err := validateUnimplemented(m); err != nil {
+		return err
+	}
+
+	if m.GetAsOf().IsUnknown() {
+		return throw.New("AsOf can not be Unknown")
+	}
+
+	if m.GetAsOf() >= currPulse {
+		return throw.New("AsOf should be less that current pulse")
+	}
+
+	if m.GetObject().IsEmpty() {
+		return throw.New("Object should not be empty")
+	}
+
+	if !m.GetObject().IsSelfScope() {
+		return throw.New("Object reference should be selfScore")
+	}
+
+	if m.GetObject().GetLocal().Pulse() >= currPulse {
+		return throw.New("Object pulse should be less that current pulse")
+	}
+
+	switch m.GetStatus() {
+	case Unknown:
+		return throw.New("Unexepected Unknown status received")
+	case Ready:
+		if m.GetUnorderedPendingCount() < 0 || m.GetUnorderedPendingCount() >= 127 {
```

Comment:
I would try move code from case's to separate functions

Reply:
Done